faux-mgs/src/main.rs

```diff
@@ -2015,6 +1962,123 @@ async fn monorail_unlock(
     Ok(())
 }
 
+fn unlock_permslip(
+    log: &Logger,
+    key_name: String,
+    challenge: UnlockChallenge,
+) -> Result<UnlockResponse> {
+    use std::process::{Command, Stdio};
+
+    let mut permslip = Command::new("permslip")
+        .arg("sign")
+        .arg(key_name)
+        .arg("--sshauth")
+        .arg("--kind=tech-port-unlock-challenge")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::inherit())
+        .spawn()
+        .context(
+            "unable to execute `permslip`, is it in your PATH and executable?",
+        )?;
```

Comment:
I'll defer to others but I have a light preference to also allow overloading the path via environment variable e.g. `PERMSLIP` if I need to test something

Reply:
Good idea, added in 68df5ec.